### PR TITLE
fix(projections): add user grant projection

### PR DESCRIPTION
**Pull request by adlerhurst**
merge after https://github.com/caos/zitadel/pull/2796

### CODE REVIEW

File: internal/query/projection/projection.go [hunk 1]
```diff
@@ -65,6 +65,7 @@ func Start(ctx context.Context, sqlClient *sql.DB, es *eventstore.Eventstore, co
 	NewProjectGrantMemberProjection(ctx, applyCustomConfig(projectionConfig, config.Customizations["project_grant_members"]))
 	_, err := NewKeyProjection(ctx, applyCustomConfig(projectionConfig, config.Customizations["keys"]), defaults.KeyConfig)
 	NewAuthNKeyProjection(ctx, applyCustomConfig(projectionConfig, config.Customizations["authn_keys"]))
+	NewUserGrantProjection(ctx, applyCustomConfig(projectionConfig, config.Customizations["user_grant"]))
```
[the remark this answers is not in the document]
Comment (livio-a) [reply internal/query/projection/projection.go:68]: for consistency we should use plural forms

File: internal/query/projection/user_grant.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
+package projection
```
Comment (livio-a) [inline internal/query/projection/user_grant.go:1]: please write tests for this file

**Review by livio-a — CHANGES_REQUESTED** [review]

File: internal/query/projection/user_grant.go [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+					Reduce: p.reduceRemoved,
+				},
+				{
+					Event:  usergrant.UserGrantDeactivatedType,
+					Reduce: p.reduceDeactivated,
+				},
+				{
+					Event:  usergrant.UserGrantReactivatedType,
+					Reduce: p.reduceReactivated,
+				},
```
Comment (fgerschwiler) [inline internal/query/projection/user_grant.go:64]: Handle user.delete and project.delete

File: internal/query/projection/user_grant.go [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+		[]handler.Condition{
+			handler.NewCond(UserGrantID, event.Aggregate().ID),
+		},
+	), nil
+}
+
+func (p *UserGrantProjection) reduceRemoved(event eventstore.Event) (*handler.Statement, error) {
+	switch event.(type) {
+	case *usergrant.UserGrantRemovedEvent, *usergrant.UserGrantCascadeRemovedEvent:
+		// ok
```
Comment (fgerschwiler) [inline internal/query/projection/user_grant.go:137]: what does this //ok mean? Event should be handled?

Comment (livio-a) [reply internal/query/projection/user_grant.go:137]: events will be handled... it's just the easiest way to catch both event types and remove events don't have any data

**Review by fgerschwiler — APPROVED** [review]

**Review by livio-a — APPROVED** [review]

**Pull request merged**